src/uct/cuda/base/cuda_md.c

```diff
@@ -127,6 +128,8 @@ uct_cuda_base_query_attributes(uct_cuda_copy_md_t *md, const void *address,
     attr_data[1] = &is_managed;
     attr_type[2] = CU_POINTER_ATTRIBUTE_DEVICE_ORDINAL;
     attr_data[2] = &cuda_device;
+    attr_type[3] = CU_POINTER_ATTRIBUTE_CONTEXT;
+    attr_data[3] = &cuda_mem_ctx;
```

Comment:
@Akshay-Venkatesh `cudaMallocAsync` has first appeared in 11.2.0, but `CU_POINTER_ATTRIBUTE_MEMPOOL_HANDLE` only appeared in 11.3.0, therefore I changed the check here which should still be equivalent, based on offline conversations with the CUDA team, see the updated code comments just below. Also, `CU_POINTER_ATTRIBUTE_CONTEXT` has existed since at least CUDA 10.0, so I believe we don't need any extra preprocessor checks here, what do you think?